File: LeetCode/java/2103_Rings_and_Rods.java
import java.util.HashMap;
import java.util.HashSet;
import java.util.Map;
import java.util.Set;

class Solution {
    public int countPoints(String rings) {
        int answer = 0;
        Map<Integer, Set<Character>> map = new HashMap<>();

        for (int i = 0; i < rings.length(); i += 2) {
            char color = rings.charAt(i);
            int num = rings.charAt(i + 1) - 'A';

            if (!map.containsKey(num)) {
                map.put(num, new HashSet<>());
            }

            map.get(num).add(color);
        }

        for (int key : map.keySet()) {
            if (map.get(key).size() == 3) {
                answer++;
            }
        }

        return answer;
    }
}

File: LeetCode/cpp/2103_Rings_and_Rods.cpp
class Solution {
public:
	int countPoints(string rings) {
		int answer = 0;
		unordered_map<int, set<char>> um;

		for (int i = 0; i < rings.length(); i += 2) {
			char color = rings[i];
			int num = rings[i + 1] - 'A';

			um[num].insert(color);
		}

		for (auto m : um) {
			if (m.second.size() == 3) {
				answer++;
			}
		}

		return answer;
	}
};